File: native/pe-padding-diversifier/src/options.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct TransformOptions {
  pub seed: u64,
  pub minimum_sled_length: usize,
  pub maximum_mutations: Option<usize>,
  pub dry_run: bool,
  pub allow_invalid_signature: bool,
  pub allow_zero_padding: bool,
  pub dll_only: bool,
}

impl Default for TransformOptions {
  fn default() -> Self {
    Self {
      seed: 0,
      minimum_sled_length: 8,
      maximum_mutations: None,
      dry_run: false,
      allow_invalid_signature: false,
      allow_zero_padding: false,
      dll_only: true,
    }
  }
}

File: internal/pepad/options.go
package pepad

// programVersion is part of candidateSeed and must stay byte-for-byte aligned
// with native/pe-padding-diversifier/Cargo.toml in the Electron source tree.
const programVersion = "0.1.0"

type Options struct {
	Seed                  uint64 `json:"seed"`
	MinimumSledLength     int    `json:"minimum_sled_length"`
	MaximumMutations      *int   `json:"maximum_mutations"`
	DryRun                bool   `json:"dry_run"`
	AllowInvalidSignature bool   `json:"allow_invalid_signature"`
	AllowZeroPadding      bool   `json:"allow_zero_padding"`
	DLLOnly               bool   `json:"dll_only"`
}

func DefaultOptions() Options {
	return Options{MinimumSledLength: 8, DLLOnly: true}
}
